documents/entity/service.go

```diff
@@ -210,13 +216,87 @@ func (s service) DeriveEntityResponse(model documents.Model) (*cliententitypb.En
 		return nil, err
 	}
 
+	entityID := model.ID()
+	var relationships []*cliententitypb.Relationship
+	// if this identity is not the owner of the entity, return an empty relationships array
+	selfDID, err := contextutil.AccountDID(ctx)
+	if err != nil {
+		return nil, errors.New("failed to get self ID")
+	}
+
+	isCollab, err := model.IsDIDCollaborator(selfDID)
+	if err != nil {
+		return nil, err
+	}
+	if !isCollab {
+		return &cliententitypb.EntityResponse{
+			Header: h,
+			Data: &cliententitypb.EntityDataResponse{
+				Entity:        data,
+				Relationships: relationships,
+			},
+		}, nil
+	}
+	_, models, err := s.ListEntityRelationships(ctx, entityID)
+	if err != nil {
+		return nil, err
+	}
+
+	//list the relationships associated with the entity
+	if models != nil {
+		for _, m := range models {
+			active := false
+			tokens, err := m.GetAccessTokens()
+			if err != nil {
+				return nil, err
+			}
+			if len(tokens) != 0 {
+				active = true
+			}
+			targetDID := m.(*entityrelationship.EntityRelationship).TargetIdentity.String()
+			r := &cliententitypb.Relationship{
+				Identity: targetDID,
+				Active:   active,
+			}
+			relationships = append(relationships, r)
+		}
+	}
+
 	return &cliententitypb.EntityResponse{
 		Header: h,
 		Data: &cliententitypb.EntityDataResponse{
 			Entity:        data,
-			Relationships: nil,
+			Relationships: relationships,
 		},
 	}, nil
+}
+
+// DeriveEntityRelationshipData returns the relationship data from an entity relationship model
+func (s service) DeriveEntityRelationshipData(model documents.Model) (*cliententitypb.RelationshipData, error) {
+	return s.erService.DeriveEntityRelationshipData(model)
+}
+
+// DeriveRelationshipsListResponse returns a relationships list response from the array of relationship models
+func (s service) DeriveRelationshipsListResponse(entity documents.Model, models []documents.Model) (*cliententitypb.RelationshipResponse, error) {
```

Comment:
we do not need this right?

Reply:
I've removed it in my PR, requested your review on Manuel's by mistake :)